bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/AccordionImpl.java

```diff
@@ -110,4 +117,34 @@ public String getHeadingElement() {
         }
         return null;
     }
+
+    /*
+     * DataLayerProvider implementation of field getters
+     */
+
+    @Override
+    public String[] getDataLayerShownItems() {
+        if (expandedItems == null) {
+            return new String[0];
+        }
+
+        if (expandedItemIds == null) {
+            List<String> expandedItemsName = Arrays.asList(expandedItems);
+            List<ListItem> items = this.getItems();
+
+            if (items != null) {
+                List<String> expandedItemIdsList = items.stream()
+                    .filter(item -> expandedItemsName.contains(item.getName()))
+                    .map(item -> item.getId())
```

Comment:
shouldn't it be instead?
`item.getDataLayer().getId() `

Reply:
It's actually the same as `DataLayer#getId` is just a wrapper on `AbstractComponent#getId`:

https://github.com/adobe/aem-core-wcm-components/pull/976/files/8bee2877af337589587ca32090fb17933b122899#diff-23a6f62806626f80ca4af8c8ca29a634R72

Reply:
But you're right, at some point we might change that and need to keep consistent.

Reply:
exactly, it works now because getId() === getDataLayer().getId() in the current implementation, but this might change one day.

Reply:
Done in 2509408